src/app/components/user/UserWorkspaces/UserWorkspacesComponent.js

```diff
@@ -0,0 +1,243 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Relay from 'react-relay/classic';
+import { FormattedMessage } from 'react-intl';
+import { commitMutation, graphql } from 'react-relay/compat';
+import { browserHistory } from 'react-router';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemAvatar from '@material-ui/core/ListItemAvatar';
+import ListItemText from '@material-ui/core/ListItemText';
+import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
+import cx from 'classnames/bind';
+import Tooltip from '../../cds/alerts-and-prompts/Tooltip';
+import NextIcon from '../../../icons/chevron_right.svg';
+import PrevIcon from '../../../icons/chevron_left.svg';
+import BlankState from '../../layout/BlankState';
+import SettingsHeader from '../../team/SettingsHeader';
+import TeamAvatar from '../../team/TeamAvatar';
+import ButtonMain from '../../cds/buttons-checkboxes-chips/ButtonMain';
+import CreateTeamDialog from '../../team/CreateTeamDialog';
+import { FlashMessageSetterContext } from '../../FlashMessage';
+import { getErrorMessageForRelayModernProblem } from '../../../helpers';
+import { stringHelper } from '../../../customHelpers';
+import styles from '../user.module.css';
+import workspaceStyles from './UserWorkspacesComponent.module.css';
+import MediasLoading from '../../media/MediasLoading';
+
+const updateUserMutation = graphql`
+  mutation UserWorkspacesComponentUpdateUserMutation($input: UpdateUserInput!) {
+    updateUser(input: $input) {
+      me {
+        current_team_id
+      }
+    }
+  }
+`;
+
+const UserWorkspacesComponent = ({
+  currentTeam,
+  numberOfTeams,
+  pageSize,
+  relay,
+  teams,
+  totalCount,
+}) => {
+  const [showCreateTeamDialog, setShowCreateTeamDialog] = React.useState(false);
+  const [cursor, setCursor] = React.useState(0);
+  const [isPaginationLoading, setIsPaginationLoading] = React.useState(false);
+  const setFlashMessage = React.useContext(FlashMessageSetterContext);
+
+  const onFailure = (errors) => {
+    const errorMessage = getErrorMessageForRelayModernProblem(errors) || <FormattedMessage
+      id="switchTeams.error"
```

Comment:
Please fix the namespace to match the component name.

Reply:
I'm pair-reviewing this with Sarah and asked to keep existing string ids to avoid generating retranslations.